### PR TITLE
update the lib also if  internet is working

### PR DESCRIPTION
**Pull request by szaimen**
I think this makes sense since if the internet works we can also update the lib (should reduce compatibility problems)

### CODE REVIEW

**Comment by enoch85** [comment]
> update the lib also if internet is working 

I don't know what checking if that file has to do with internet?

**Comment by szaimen** [comment]
> > update the lib also if internet is working 
> 
> I don't know what checking if that file has to do with internet?

I can explain it to you:

If you are so far in the startup script that the you cannot run the startup script twice file is created, you have a working internet connection. That is the reason why I think it makes sense to also update the lib if this file exists. Otherwise you will always don't update the lib during the whole startup script which can cause complications if you run scripts directly from the repository which may need a newer lib.

**Comment by szaimen** [comment]
We could alternatively just download a new library once while downloading all other scripts which might be a better approach...

**Comment by enoch85** [comment]
Actually, in the new official VM you don't need an internet connection at all to run the first setup. I think that should be the default in this as well. 

I'm feeling more and more confident that `fetch_lib.sh` will work in all scripts now, especially since we added it to the startup script.

But sure, we can merge this for now.

**Comment by szaimen** [comment]
Please wait I would like to change the implementation a bit

**Comment by szaimen** [comment]
I think this is done now

File: nextcloud-startup-script.sh [hunk 1]
```diff
@@ -217,6 +217,8 @@ download_script MENU menu
 download_script MENU server_configuration
 download_script MENU nextcloud_configuration
 download_script MENU additional_apps
+# This is needed to update the lib once during the startup script
+download_script GITHUB_REPO lib
```
Comment (enoch85) [inline nextcloud-startup-script.sh:221]: This is one step away from my plan with running the first startup without internet. Please revert.

Comment (enoch85) [reply nextcloud-startup-script.sh:221]: The other solution was better.

Comment (szaimen) [reply nextcloud-startup-script.sh:221]: If your plan is to run the whole startup script without internet, is the current implementation in master the best.
But how did you plan to get rid of run_script in every menu? 

I would suggest cloning the whole repo furing the install_script and changing run_script and download_script to use the local script if existing.

Comment (enoch85) [reply nextcloud-startup-script.sh:221]: > run_script in every menu?

Haven't figured that out yet, but maybe skip the whole menu section if the user doesn't have internet. They can always run it later.

**Comment by enoch85** [comment]
Closing as we decided that the current master is better,

File: nextcloud-startup-script.sh (hunk 1 above)
Comment (szaimen) [reply nextcloud-startup-script.sh:221]: Of course this is also a solution, yes 👍
It might also be much easier to implement

**Pull request closed without merging**